lms/djangoapps/course_structure_api/v0/views.py

```diff
@@ -0,0 +1,233 @@
+        if request.user.is_anonymous() and not settings.DEBUG:
+            raise AuthenticationFailed
+
+
+class CourseList(CourseViewMixin, ListAPIView):
+    """
+    **Use Case**
+
+        CourseList returns paginated list of courses in the edX Platform. The list can be filtered by course_id.
+
+    **Example Request**
+
+          GET /api/course_structure/v0/courses/
+          GET /api/course_structure/v0/courses/?course_id={course_id1},{course_id2}
+
+    **Response Values**
+
+        * id: The unique identifier for the course.
+
+        * name: The name of the course.
+
+        * category: The type of content. In this case, the value is always "course".
+
+        * org: The organization specified for the course.
+
+        * course: The course number.
+
+        * org: The run for the course.
+
+        * uri: The URI to use to get details of the course.
+
+        * image_url: The URI for the course's main image.
+
+        * start: Course start date
+
+        * end: Course end date
+    """
+    paginate_by = 10
+    paginate_by_param = 'page_size'
+    pagination_serializer_class = PaginationSerializer
+    serializer_class = serializers.CourseSerializer
+
+    def get_queryset(self):
+        course_ids = self.request.QUERY_PARAMS.get('course_id', None)
+
+        course_descriptors = []
+        if course_ids:
+            course_ids = course_ids.split(',')
+            for course_id in course_ids:
+                course_key = CourseKey.from_string(course_id)
+                course_descriptor = courses.get_course(course_key)
+                course_descriptors.append(course_descriptor)
+        else:
+            course_descriptors = modulestore().get_courses()
+
+        results = [course for course in course_descriptors if self.user_can_access_course(self.request.user, course)]
+
+        # Sort the results in a predictable manner.
+        results.sort(key=attrgetter('id'))
+
+        return results
+
+
+class CourseDetail(CourseViewMixin, RetrieveAPIView):
+    """
+    **Use Case**
+
+        CourseDetail returns details for a course.
+
+    **Example requests**:
+
+        GET /api/course_structure/v0/courses/{course_id}/
+
+    **Response Values**
+
+        * category: The type of content.
+
+        * name: The name of the course.
+
+        * uri: The URI to use to get details of the course.
+
+        * course: The course number.
+
+        * due:  The due date. For courses, the value is always null.
+
+        * org: The organization specified for the course.
+
+        * id: The unique identifier for the course.
+    """
+
+    serializer_class = serializers.CourseSerializer
+
+    def get_object(self, queryset=None):
+        return self.get_course_or_404()
+
+
+class CourseStructure(CourseViewMixin, RetrieveAPIView):
+    """
+    **Use Case**
+
+        Retrieves course structure.
+
+    **Example requests**:
+
+        GET /api/course_structure/v0/courses/{course_id}/structure/
```

Comment:
To clarify a bit more, the API guidelines suggest that the path be something like `/api/course_structure/v0/course_structure/{course_id}`. The reasoning is that having this be a suffix on another resource implies that this is somehow a subordinate resource, when it is in fact something else altogether.